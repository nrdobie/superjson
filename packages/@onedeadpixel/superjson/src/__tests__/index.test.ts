import SJSON from '..'

test('SJSON should hand complex data', () => {
  const complex = {
    string: 'hello world',
    number: 1234,
    boolean: true,
    date: new Date('1995-12-04T00:00:00Z'),
    maps: new Map([
      [1, 'Mocha'],
      [2, 'LiveScript'],
      [3, 'JavaScript'],
      [4, 'ECMAScript']
    ]),
    sets: new Set(['Mocha', 'LiveScript', 'JavaScript', 'ECMAScript'])
  }

  const str = SJSON.stringify(complex, null, '  ')

  expect(str).toMatchSnapshot()

  const parsed = SJSON.parse(str)

  expect(parsed).toEqual(complex)
})
