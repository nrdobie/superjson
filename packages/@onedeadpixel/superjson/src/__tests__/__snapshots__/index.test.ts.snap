// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SJSON should hand complex data 1`] = `
"{
  \\"string\\": \\"hello world\\",
  \\"number\\": 1234,
  \\"boolean\\": true,
  \\"date\\": {
    \\"__sj_type\\": \\"Date\\",
    \\"__sj_value\\": \\"1995-12-04T00:00:00.000Z\\"
  },
  \\"maps\\": {
    \\"__sj_type\\": \\"Map\\",
    \\"__sj_value\\": [
      [
        1,
        \\"Mocha\\"
      ],
      [
        2,
        \\"LiveScript\\"
      ],
      [
        3,
        \\"JavaScript\\"
      ],
      [
        4,
        \\"ECMAScript\\"
      ]
    ]
  },
  \\"sets\\": {
    \\"__sj_type\\": \\"Set\\",
    \\"__sj_value\\": [
      \\"Mocha\\",
      \\"LiveScript\\",
      \\"JavaScript\\",
      \\"ECMAScript\\"
    ]
  }
}"
`;
